tests/lax_numpy_test.py

```diff
@@ -893,19 +893,21 @@ def np_fun(lhs, rhs):
   @jtu.sample_product(
     dtype=[dt for dt in float_dtypes if dt not in [jnp.float16, jnp.bfloat16]],
     shape=[shape for shape in one_dim_array_shapes if shape != (1,)],
+    num_rhs=[1, 5],
     deg=[1, 2, 3],
     rcond=[None, -1, 10e-3, 10e-5, 10e-10],
     full=[False, True],
     w=[False, True],
     cov=[False, True, "unscaled"],
   )
   @jax.default_matmul_precision("float32")
-  def testPolyfit(self, shape, dtype, deg, rcond, full, w, cov):
+  def testPolyfit(self, shape, num_rhs, dtype, deg, rcond, full, w, cov):
     rng = jtu.rand_default(self.rng())
     tol_spec = {np.float32: 1e-3, np.float64: 1e-13, np.complex64: 1e-5}
     tol = jtu.tolerance(dtype, tol_spec)
     _w = lambda a: abs(a) if w else None
-    args_maker = lambda: [rng(shape, dtype), rng(shape, dtype), rng(shape, dtype)]
+    rhs_shape = shape + (num_rhs,) if num_rhs > 1 else shape
+    args_maker = lambda: [rng(shape, dtype), rng(rhs_shape, dtype), rng(rhs_shape, dtype)]
```

Comment:
When I run the tests on this branch locally, I find that this logic doesn't seem correct. The weights need to be 1d so I think this should be:

```suggestion
    rhs_shape = shape + (num_rhs,) if num_rhs > 1 else shape
    args_maker = lambda: [rng(shape, dtype), rng(rhs_shape, dtype), rng(shape, dtype)]
```

You can run the tests locally using:

```bash
python -m pytest -v tests/lax_numpy_test.py -k "testPolyFit"
```

Do they work for you as written?

Reply:
No I'm sorry, I should have marked this as a draft after seeing those tests fail. The broadcasting wasn't right, and I though I'd need to account for something else.

I didn't make the weights shape `shape` since I thought the weights would have shape `(M, K)`, but now I see that's wrong